blueprint/api.apib

```diff
@@ -726,114 +854,207 @@ A user is able to have multiple roles through their user roles.
             Authorization: Bearer {token}
 
-+ Response 200 (application/json)
++ Response 201 (application/vnd.api+json; charset=utf-8)
 
     + Attributes (Role Skill Response)
 
-+ Response 401 (application/json)
++ Response 401 (application/vnd.api+json; charset=utf-8)
 
     + Attributes (JSON Web Token Invalid Response)
 
-+ Response 403 (application/json)
++ Response 403 (application/vnd.api+json; charset=utf-8)
 
     + Attributes (Forbidden Response)
 
-+ Response 422 (application/json)
++ Response 422 (application/vnd.api+json; charset=utf-8)
 
     + Attributes (Unprocessable Entity Response)
 
-# Group Skills
+### List all role skills [GET]
 
-This endpoint retrieves skills that can be applied to both users (through user-skills) and roles (through role-skills)
++ Request
 
-## Skills [/skills]
+    + Headers
+
+            Accept: application/vnd.api+json
 
-Skills can be filtered by the id of the skill. This should be an array of one or multiple ids.
++ Response 200 (application/vnd.api+json; charset=utf-8)
 
-### Filter by id [GET /skills{?filter}]
+    + Attributes (Role Skills Response)
+
+## Role Skill [/role-skills/{id}]
+
+### Retrieve a role skill [GET]
 
 + Parameters
 
-    + filter: `1` (enum[number], optional) - Id query to filter returned skills
+    + id (number) - Id of a role skill.
 
 + Request
 
     + Headers
 
-            Accept: application/json
+            Accept: application/vnd.api+json
 
-+ Response 200 (application/json)
++ Response 200 (application/vnd.api+json; charset=utf-8)
 
-    + Attributes (Skills Response)
+    + Attributes (Role Skill Response)
 
-### Autocomplete skill with query [GET /skills{?query}]
++ Response 404 (application/vnd.api+json; charset=utf-8)
 
-Skills can be queried by a substring of the name of the skill. This should be a string with one or more letters of the name of the skill requested.
+    + Attributes (Record Not Found Response)
 
-For example, if we wanted to return skills like "Java" and "JavaScript", we may query with the substring "jav". A good use for this API would be a skill typeahed input.
+### Delete a role skill [DELETE]
 
 + Parameters
 
-    + query: `jav` (string, optional) - Text query to filter returned skills
+    + id (number) - Id of a role skill.
 
 + Request
 
     + Headers
 
-            Accept: application/json
+            Accept: application/vnd.api+json
+            Authorization: Bearer {token}
 
-+ Response 200 (application/json)
++ Response 204 (application/vnd.api+json; charset=utf-8)
 
-    + Attributes (Skills Response)
+    + Attributes (No Content Response)
+
++ Response 401 (application/vnd.api+json; charset=utf-8)
+
+    + Attributes (JSON Web Token Invalid Response)
+
++ Response 403 (application/vnd.api+json; charset=utf-8)
+
+    + Attributes (Forbidden Response)
+
+# Group Skills
+
+This endpoint retrieves Skills that can belong to Projects, Roles, or Users. This helps to match users to relevant projects or roles.
```

Comment:
Saying it belongs to would imply it's a one to many relationship. I'd say go with "that can relate to", or if that's not enough, be more verbose about those relations.

Example of more verbosity:

```
Skills relate to projects in the context of "some of these skills are needed for someone to work on this project".

They relate to roles in the context of "to fulfil a role, a user needs to possess one or more skills. Having a skill means they can fulfil one or more roles."

They relate to users in the context that a user can have one or more skills. Multiple users can have the same skills.